spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/flyway/FlywayAutoConfigurationTests.java

```diff
@@ -189,7 +189,7 @@ public void overrideSchemas() {
 	@Test
 	public void changeLogDoesNotExist() {
 		this.contextRunner.withUserConfiguration(EmbeddedDataSourceConfiguration.class)
-				.withPropertyValues("spring.flyway.locations:file:no-such-dir")
+				.withPropertyValues("spring.flyway.locations:filesystem:no-such-dir")
```

Comment:
Can we keep a test that uses `file`?

Reply:
I guess I didn't feel like there was any value in testing a `file:` location, since it would just fail in Flyway anyway because of the unknown prefix, regardless of whether it existed or not.

Happy to add it back in if you want though..?

Reply:
Spring users know about `file` and `classpath` so it would be nice to offer that as a convenience. But I guess the swapping from `file` to `filesystem` starts to be a bit funky so let's keep it as is for now. Thanks.

Reply:
Agreed. I did consider swapping `file` to `filesystem` but it felt a bit dirty. And I think the error from Flyway in this case is pretty clear anyway:

    Unknown prefix for location (should be either filesystem: or classpath:): file:src/main/resources/db/migration